core/base/test/TStringTest.cxx

```diff
@@ -0,0 +1,15 @@
+#include "gtest/gtest.h"
+#include "gmock/gmock.h"
+
+#include "TString.h"
+
+TEST(TString, Basics)
+{
+   TString n("Test", -5);
```

Comment:
we should also capture the Error, otherwise it will end up in our test logs.

you need to `#include "ROOTUnitTestSupport.h"` and then you can execute code that should error out such as this and also line 13 within `ROOT_EXPECT_ERROR`, e.g.

```cpp
TString *s = nullptr;
ROOT_EXPECT_ERROR(s = new TString("Test", -5), "TString::TString", "<error message goes here>");
```